pkg/controller/servicebindingrequest/annotations_test.go

```diff
@@ -105,7 +105,55 @@ func TestAnnotationsSetAndRemoveSBRAnnotations(t *testing.T) {
 	objs = append(objs, u)
 
 	t.Run("SetSBRAnnotations", func(t *testing.T) {
-		err := SetSBRAnnotations(client, namespacedName, objs)
+		originCopy := u.DeepCopy()
+		newObj := SetSBRAnnotations(namespacedName, u)
+
+		// we are not modifying the origin object
+		equal, err := nestedMapComparison(u, originCopy)
+		require.NoError(t, err)
+		require.True(t, equal)
+
+		equal, err = nestedMapComparison(u, newObj, []string{"metadata", "annotations"}...)
+		require.NoError(t, err)
+		require.False(t, equal)
+
+		objNamespacedName, err := GetSBRNamespacedNameFromObject(newObj)
+		require.NoError(t, err)
+		require.Equal(t, namespacedName, objNamespacedName)
+
+		// assert nothing else is changed
+		newObj.SetAnnotations(nil)
+		equal, err = nestedMapComparison(u, newObj)
+		require.NoError(t, err)
+		require.True(t, equal)
+	})
+
+	t.Run("RemoveSBRAnnotations", func(t *testing.T) {
+		originCopy := u.DeepCopy()
```

Comment:
Perhaps `originCopy` can be simplified since `RemoveSBRAnnotations` is already doing a deep copy.